### PR TITLE
Adding support for Anizium

### PR DESCRIPTION
**Pull request by Fen1kks**
This pull request adds support for Anizium.
- Implemented main.ts for Anizium
- Added style.less for design
- Added tests.json for functionality testing

### CODE REVIEW

**Comment by Source-Dom** [comment]
Hey iirc you're not supposed to touch playerUrls.js

**Comment by Fen1kks** [comment]
> Hey iirc you're not supposed to touch playerUrls.js

Instead of using HTML5, the site uses x.anizium.co/embed, i.e. an iframe, so it is necessary for the player to be detectable.

**Comment by henrik9999** [comment]
Revert the last commit, keep the test failing

**Comment by henrik9999** [comment]
Nvm keep it like it is if the site requires login

**Comment by Fen1kks** [comment]
Yes, the site requires an account.

**Comment by Fen1kks** [comment]
I don't understand what I should do now.

**Comment by henrik9999** [comment]
Add the offline flag again I thought it didnt require login

**Comment by Fen1kks** [comment]
@henrik9999 While seasons are listed on the site as Season 1, Season 2, etc., some series like Dr. Stone have been given separate names, so I don't know how to optimize this. It automatically finds only the first season. Even if I open the second season, it still acts as if I'm in the first season and asks, “Are you watching it again?”

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import { PageInterface } from '../../pageInterface';
+
+export const Anizium: PageInterface = {
+  name: 'Anizium',
+  type: 'anime',
+  domain: 'https://anizium.co',
+  languages: ['Turkish'],
+  urls: {
+    match: ['*://anizium.co/*', '*://www.anizium.co/*', '*://m.anizium.co/*'],
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: I think m.anizium.co does not exist. You can combine the other two to one by using '*://*.anizium.co/*'

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  },
+  search: 'https://anizium.co/search?q={searchtermPlus}',
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: I think this should be `https://anizium.co/search?value={searchtermPlus}`

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      return $c
+        .coalesce(
+          // Primary: trending-text selector with cleaning
+          element.text().trim().replaceRegex('\\s+S\\d+\\s+B\\d+.*', '').trim().run(),
+          // Fallback 1: page title cleaning
+          $c.title().replaceRegex(' - Anizium.*', '').trim().run(),
+          // Fallback 2: meta og:title
+          $c.querySelector('meta[property="og:title"]').getAttribute('content').run(),
+          // Fallback 3: h1 title
+          $c.querySelector('h1').text().trim().run(),
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Whats up with all these random fallbacks? And still returns a wrong title

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          baseId.concat('?s=').concat(season.string().run()).run(),
+          // Fallback: base ID only
+          baseId.run(),
+        )
+        .run();
+    },
+    getOverviewUrl($c) {
+      // Get base ID (without season parameter)
+      const baseId = $c.url().urlPart(4);
+      return $c.string('https://anizium.co/anime/').concat(baseId.run()).run();
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Better use `$c.string('/anime/').concat(baseId.run()).urlAbsolute().run();` then we do not have an hardcoded domain

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        .run();
+    },
+  },
+  overview: {
+    isOverviewPage($c) {
+      return $c.url().contains('/anime/').run();
+    },
+    getTitle($c) {
+      return $c
+        .coalesce(
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Same as sync.getTitle

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+          $c.querySelector('meta[property="og:title"]').getAttribute('content').run(),
+        )
+        .ifNotReturn()
+        .run();
+    },
+    getIdentifier($c) {
+      return $c.url().urlPart(4).run();
+    },
+    uiInjection($c) {
+      return $c.querySelector('h5.main-title').uiAfter().run();
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts:112]: Does not seem to work

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    },
+  },
+  list: {
+    elementsSelector($c) {
+      return $c.querySelectorAll('a[href*="watch"]').run();
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Matches too many links better use `.episode-block`

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      return $c
+        .coalesce(
+          // Primary: URL parameter
+          $c.url().urlParam('episode').number().run(),
+          // Fallback 1: from page elements
+          $c.querySelector('.episode-info').text().regex('(\\d+)', 1).number().run(),
+          // Fallback 2: from breadcrumb or navigation
+          $c.querySelector('.breadcrumb').text().regex('Episode\\s+(\\d+)', 1).number().run(),
+          // Default fallback
+          $c.number(1).run(),
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Is there a case where there is no episode urlParam?

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+  },
+  list: {
+    elementsSelector($c) {
+      return $c.querySelectorAll('a[href*="watch"]').run();
+    },
+    elementUrl($c) {
+      return $c.getAttribute('href').urlAbsolute().run();
+    },
+    elementEp($c) {
+      return $c.getAttribute('href').regex('[&?]episode=(\\d+)', 1).number().run();
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Why not use .urlParam('episode') like in getEpisode?

File: src/pages-chibi/implementations/Anizium/tests.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      "url": "https://m.anizium.co/anime/777888999",
+      "expected": {
+        "sync": false,
+        "title": "Mobile Anime Overview",
+        "identifier": "777888999",
+        "uiSelector": true
+      }
+    }
+  ]
+}
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/tests.json]: Please remove offline and replace with real data and not only placeholder data. Min 1 overview and 1 sync. Additional tests for interessting/edge cases.

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Comment by Fen1kks** [comment]
@lolamtisch Is there anything else you want me to do?

File: src/pages-chibi/implementations/Anizium/main.ts [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+      return $c
+        .if(
+          nextButton.getAttribute('href').boolean().run(),
+          nextButton.getAttribute('href').urlAbsolute().run(),
+          $c.string('').run(),
+        )
+        .run();
+    },
+  },
+  overview: {
```
Comment (lolamtisch) [inline src/pages-chibi/implementations/Anizium/main.ts]: Overview implementation still has some things broken. 

But more important I just noticed that the overview has a season dropdown. Meaning that the overview implementation is too simple either way. You could just drop the overview implementation completely if you are lazy. Or handling dropdown changes, so that when the user changes the dropdown it is like a new website and triggers an update with `"identifier": "392217205?s=2"` etc.

> While seasons are listed on the site as Season 1, Season 2, etc., some series like Dr. Stone have been given separate names, so I don't know how to optimize this. It automatically finds only the first season. Even if I open the second season, it still acts as if I'm in the first season and asks, “Are you watching it again?”

There is no real fix for that. What we normally do is that from season 2 we just add "Season X" to the end of the title. It at least works sometimes.

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Review by lolamtisch — DISMISSED** [review]
